### PR TITLE
Adding new fields for the event creation

### PR DESCRIPTION
**Pull request by Tikipiou**
# Event creation

## Overview

Design review, added more fields and code cleaning.

## Modifications

- Updated the `CreateEvent` component with a floating buttons and list of fields.
- Updated the `EventViewModel` for the new functionalities.
- Updated the `DbService`and `DbServiceImpl` to update the firestore document.
- Created the `DatePickers` for code clarity
- Created the `ImageListItem` for modularity
- Created the `Event.Field` type with `Text` and `Image` subtypes

## Testing

- Updated the `CreateEventTest`.

## Dependencies

- No additional dependencies required.

### CODE REVIEW

File: app/src/androidTest/java/com/swent/assos/CreateEventTest.kt [hunk 1]
```diff
@@ -84,7 +83,6 @@ class CreateEventTest : SuperTest() {
     run {
       ComposeScreen.onComposeScreen<CreateEventScreen>(composeTestRule) {
         step("Fill the form") {
-          // form { assertIsDisplayed() }
```
Comment (marinPh) [inline app/src/androidTest/java/com/swent/assos/CreateEventTest.kt:87]: either the line is needed or you can delete it

File: app/src/main/java/com/swent/assos/ui/components/EventItem.kt [hunk 2]
```diff
@@ -34,8 +33,7 @@ fun EventItem(event: Event, navigationActions: NavigationActions, asso: Associat
       colors = CardDefaults.cardColors(MaterialTheme.colorScheme.onPrimary),
       shape = RoundedCornerShape(12.dp),
       modifier =
-          Modifier.testTag("NewsItem")
```
Comment (marinPh) [inline app/src/main/java/com/swent/assos/ui/components/EventItem.kt:37]: We don't need the testtag anymore

File: app/src/main/java/com/swent/assos/ui/components/EventItem.kt [hunk 3]
```diff
@@ -44,10 +42,7 @@ fun EventItem(event: Event, navigationActions: NavigationActions, asso: Associat
                 navigationActions.navigateTo(Destinations.EVENT_DETAILS.route + "/${event.id}")
               }) {
         Column(
-            modifier =
-                Modifier.width(200.dp).padding(vertical = 0.dp).clickable(true) {
-                  navigationActions.navigateTo(Destinations.EVENT_DETAILS.route + "/${event.id}")
-                },
+            modifier = Modifier.width(200.dp).padding(vertical = 0.dp),
```
Comment (marinPh) [inline app/src/main/java/com/swent/assos/ui/components/EventItem.kt:45]: are we sure it is card that we want to click not the column?

**Review by marinPh — CHANGES_REQUESTED** [review]
A wonderful PR, after testing it manually, really adds a lot to how users can create events. The UI is well done and very pretty. Most of all it is very user-friendly, well done

File: app/src/main/java/com/swent/assos/ui/components/EventItem.kt (hunk 3 above)
Comment (Tikipiou) [reply app/src/main/java/com/swent/assos/ui/components/EventItem.kt:45]: Yes and it resolved the click bug

**Review by marinPh — APPROVED** [review]
All changes required were corrected thank you

**Pull request merged**